src/webhook/handlers/processImage.js

```diff
@@ -0,0 +1,223 @@
+import { t, msgid, ngettext } from 'ttag';
+import {
+  getLineContentProxyURL,
+  createPostbackAction,
+  POSTBACK_NO_ARTICLE_FOUND,
+} from './utils';
+import gql from 'src/lib/gql';
+import ga from 'src/lib/ga';
+
+export default async function({ data = {} }, event, userId) {
+  const proxyUrl = getLineContentProxyURL(event.messageId);
+  // console.log(`Image url:  ${proxyUrl}`);
+
+  const visitor = ga(userId, '__PROCESS_IMAGE__', proxyUrl);
+  // Track text message type send by user
+  visitor.event({ ec: 'UserInput', ea: 'MessageType', el: event.message.type });
+
+  let replies;
+  data = {
+    // Start a new session
+    sessionId: Date.now(),
+  };
+
+  const {
+    data: { ListArticles },
+  } = await gql`
+    query($mediaUrl: String!) {
+      ListArticles(
+        filter: { mediaUrl: $mediaUrl }
+        orderBy: [{ _score: DESC }]
+        first: 4
+      ) {
+        edges {
+          score
+          node {
+            id
+            articleType
+            attachmentUrl(variant: THUMBNAIL)
+            attachmentHash
+          }
+        }
+      }
+    }
+  `({
+    mediaUrl: proxyUrl,
+  });
+
+  if (ListArticles.edges.length) {
+    // Track if find similar Articles in DB.
+    visitor.event({ ec: 'UserInput', ea: 'ArticleSearch', el: 'ArticleFound' });
+
+    // Track which Article is searched. And set tracking event as non-interactionHit.
+    ListArticles.edges.forEach(edge => {
+      visitor.event({
+        ec: 'Article',
+        ea: 'Search',
+        el: edge.node.id,
+        ni: true,
+      });
+    });
+
+    const articleOptions = ListArticles.edges
+      .map(({ node: { attachmentUrl, id }, score }, index) => {
+        const imgNumber = index + 1;
+        const displayTextWhenChosen = ngettext(
+          msgid`No.${imgNumber}`,
+          `No.${imgNumber}`,
+          imgNumber
+        );
+
+        const similarity = score - 1;
```

Comment:
Let's add a comment denoting the reason why we -1:

```suggestion
        // ListArticle score is 1~2 for the current query; the variable part is the ID hash difference
        const similarity = score - 1;
```